applications/vanilla/views/modules/discussionssortfilter.php

```diff
@@ -2,19 +2,42 @@
 <div class="ButtonGroup discussion-sort-filter-module pull-left">
     <?php if ($this->showSorts()) { ?>
         <span class="discussion-sorts">
-        <?php foreach ($this->getSortData() as $sort) {
-            echo anchor(val('name', $sort), val('url', $sort), [
-                    'rel' => val('rel', $sort),
-                    'class' => 'btn-default Button NavButton SortButton ' . val('cssClass', $sort, '')
-                ]) . ' ';
-        }
+        <?php
+            $sortLinks = [];
+            $foundActiveLink = false;
+
+            foreach ($this->getSortData() as $sort) {
+                $isActive = val('active', $sort);
+                $sortLinks[] = [
+                    'name' => val('name', $sort),
+                    'url' => url('/'.val('url', $sort)),
+                    'active' => $isActive,
+                ];
+
+                if ($isActive) {
+                    $foundActiveLink = true;
+                }
+            }
+
+            if(!$foundActiveLink) {
+                $sortLinks[0]['active'] = true;
+            }
+
+            echo linkDropDown($sortLinks, 'selectBox-ideationFilter', 'Sort');
         ?>
         </span>
     <?php }
     if ($this->showFilters()) { ?>
         <span class="discussion-filters">
-        <?php foreach ($this->getFilterDropdowns() as $dropdown) {
-            echo $dropdown;
+        <?php
+        foreach ($this->getFilterDropdowns() as $dropdown) {
+            $label = val('name', val(0, $dropdown), "Filter");
```

Comment:
Stick with single-quoting strings, unless you need to expand variables, etc (although I believe this was an oversight, since you're single-quoting everywhere else).